core/src/main/java/jenkins/security/BasicHeaderProcessor.java

```diff
@@ -65,7 +64,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         HttpServletResponse rsp = (HttpServletResponse) response;
         String authorization = req.getHeader("Authorization");
 
-        if (StringUtils.startsWithIgnoreCase(authorization, "Basic ")) {
+        if (authorization != null && authorization.toLowerCase().startsWith("Basic ".toLowerCase())) {
```

Comment:
While it is unlikely to matter in practice, this is _not_ a sound replacement. If a request was sent with

```
AUTHORIZATION: BASIC czNjcjN0
```

then a server running in Turkish locale would fail this clause because

```java
"basıc cznjcjn0".startsWith("basic ")
```

is false.

In general, you always want to pass the `Locale` argument (almost always `Locale.ROOT`) to `toLowerCase` / `toUpperCase`.

Reply:
@BobDu Is it worth filing a follow-up PR to address the above feedback? Or do we think the code is fine as-is?

Reply:
I agree always add `Locale.ROOT` arg to `toLowerCase` method is a best practices.
Already make a new PR #9162